prow/config/config_test.go

```diff
@@ -3779,15 +3805,267 @@ func TestRerunAuthConfigsGetRerunAuthConfig(t *testing.T) {
+			},
+			refs:     &prowapi.Refs{Org: "istio", Repo: "istio"},
+			cluster:  "",
+			expected: &prowapi.RerunAuthConfig{GitHubUsers: []string{"airbender"}},
+		},
+		{
+			name: "cluster returns matching cluster",
+			configs: []*DefaultRerunAuthConfigEntry{
+				&DefaultRerunAuthConfigEntry{
+					OrgRepo: "istio",
+					Cluster: "",
+					Config:  &prowapi.RerunAuthConfig{GitHubUsers: []string{"scrappydoo"}},
+				},
+				&DefaultRerunAuthConfigEntry{
+					OrgRepo: "istio",
+					Cluster: "trusted",
+					Config:  &prowapi.RerunAuthConfig{GitHubUsers: []string{"airbender"}},
+				},
+			},
+			refs:     &prowapi.Refs{Org: "istio", Repo: "istio"},
+			cluster:  "trusted",
+			expected: &prowapi.RerunAuthConfig{GitHubUsers: []string{"airbender"}},
+		},
+		{
+			name: "cluster returns wild card",
+			configs: []*DefaultRerunAuthConfigEntry{
+				&DefaultRerunAuthConfigEntry{
+					OrgRepo: "istio",
+					Cluster: "*",
+					Config:  &prowapi.RerunAuthConfig{GitHubUsers: []string{"scrappydoo"}},
+				},
+				&DefaultRerunAuthConfigEntry{
+					OrgRepo: "istio",
+					Cluster: "cluster",
+					Config:  &prowapi.RerunAuthConfig{GitHubUsers: []string{"airbender"}},
+				},
+			},
+			refs:     &prowapi.Refs{Org: "istio", Repo: "istio"},
+			cluster:  "trusted",
+			expected: &prowapi.RerunAuthConfig{GitHubUsers: []string{"scrappydoo"}},
+		},
+		{
+			name: "no refs with cluster returns overriding matching cluster",
+			configs: []*DefaultRerunAuthConfigEntry{
+				&DefaultRerunAuthConfigEntry{
+					OrgRepo: "",
+					Cluster: "*",
+					Config:  &prowapi.RerunAuthConfig{GitHubUsers: []string{"clarketm"}},
+				},
+				&DefaultRerunAuthConfigEntry{
+					OrgRepo: "",
+					Cluster: "trusted",
+					Config:  &prowapi.RerunAuthConfig{GitHubUsers: []string{"airbender"}},
+				},
+			},
+			refs:     nil,
+			cluster:  "trusted",
+			expected: &prowapi.RerunAuthConfig{GitHubUsers: []string{"airbender"}},
+		},
+		{
+			name: "no matching orgrepo or cluster",
+			configs: []*DefaultRerunAuthConfigEntry{
+				&DefaultRerunAuthConfigEntry{
+					OrgRepo: "notIstio",
+					Cluster: "notTrusted",
+					Config:  &prowapi.RerunAuthConfig{GitHubUsers: []string{"clarketm"}},
+				},
+			},
+			refs:     &prowapi.Refs{Org: "istio", Repo: "istio"},
+			cluster:  "trusted",
+			expected: nil,
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
+			d := Deck{}
+			d.DefaultRerunAuthConfigs = tc.configs
+			d.FinalizeDefaultRerunAuthConfigs()
 
-			if actual := tc.configs.GetRerunAuthConfig(tc.refs); !reflect.DeepEqual(actual, tc.expected) {
-				t.Errorf("Expected %v, got %v", tc.expected, actual)
+			if actual := d.GetRerunAuthConfig(tc.refs, tc.cluster); actual != nil && tc.expected != nil {
+				if !reflect.DeepEqual(*actual, *tc.expected) {
+					t.Errorf("Expected %v, got %v", tc.expected, actual)
+				}
+			} else {
+				if !reflect.DeepEqual(actual, tc.expected) {
+					t.Errorf("Expected %v, got %v", tc.expected, actual)
```

Comment:
Nit: Use `cmp.Diff` to make it easier to understand when this fails

Reply:
fixed!